vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -400,7 +402,14 @@ private void sendFile(RoutingContext context, String file, FileProps fileProps)
         wrapInTCCLSwitch(() -> {
           // guess content type
           String contentType = MimeMapping.getMimeTypeForFilename(file);
+          Set<String> contentTypesFromSuffixList = new HashSet<>();
+          for(String suffix : compressedFileSuffixes) {
+            contentTypesFromSuffixList.add(MimeMapping.getMimeTypeForExtension(suffix));
+          }
           if (contentType != null) {
+            if(compressedContentTypes.contains(contentType) || contentTypesFromSuffixList.contains(contentType)) {
```

Comment:
You shouldn't create `contentTypesFromSuffixList`. Instead, you should test if the file suffix is in `compressedFileSuffixes` or, if the file media type is in `compressedContentTypes`

Reply:
Should I use an utility method for this? 
Currently, I am finding the file suffix using the code similar to the one in ```getMimeTypeForFilename()``` method of ```MimeMapping``` class.